src/hydration/react.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react'
+import {
+  useQueryCache,
+  ReactQueryCacheProvider as CacheProvider,
+} from 'react-query'
+import { hydrate } from './hydration'
+
+import type { HydrateConfig } from './hydration'
+import type { ReactQueryCacheProviderProps } from '../react'
+
+export function useHydrate(queries: unknown, hydrateConfig?: HydrateConfig) {
+  const queryCache = useQueryCache()
+
+  // Running hydrate again with the same queries is safe,
+  // it wont overwrite or initialize existing queries,
+  // relying on useMemo here is only a performance optimization
+  React.useMemo(() => {
+    if (queries) {
+      hydrate(queryCache, queries, hydrateConfig)
+    }
+    return undefined
+  }, [queryCache, queries, hydrateConfig])
+}
+
+interface HydratorProps {
+  initialQueries?: unknown
+  hydrationConfig?: HydrateConfig
+}
+
+const Hydrator: React.FC<HydratorProps> = ({
+  initialQueries,
+  hydrationConfig,
+  children,
+}) => {
+  useHydrate(initialQueries, hydrationConfig)
+  return children as React.ReactElement<any>
+}
+
+export interface HydrationCacheProviderProps
+  extends ReactQueryCacheProviderProps {
+  initialQueries?: unknown
```

Comment:
Might be good to have a more generic name for this like `dehydratedState`. The content should be opaque to the user because we might add other things besides just queries

Reply:
That's a good point. I'll rename it and will rename the `DehydratedQueries`-type and anywhere in the docs that `dehydratedQueries` is used as a variable name as well.

Reply:
Fixed.